sentry_sdk/scope.py

```diff
@@ -146,21 +160,178 @@ class Scope(object):
+
+        final_scope = copy(_global_scope) if _global_scope is not None else Scope()
+        final_scope._type = ScopeType.MERGED
+
+        isolation_scope = _isolation_scope.get()
+        if isolation_scope is not None:
+            final_scope.update_from_scope(isolation_scope)
+
+        current_scope = _current_scope.get()
+        if current_scope is not None:
+            final_scope.update_from_scope(current_scope)
+
+        if additional_scope is not None:
+            if callable(additional_scope):
+                additional_scope(final_scope)
+            else:
+                final_scope.update_from_scope(additional_scope)
+
+        elif additional_scope_kwargs:
+            final_scope.update_from_kwargs(**additional_scope_kwargs)
+
+        return final_scope
+
+    @classmethod
+    def get_client(cls):
+        # type: () -> sentry_sdk.client.BaseClient
+        """
+        Returns the currently used :py:class:`sentry_sdk.Client`.
+        This checks the current scope, the isolation scope and the global scope for a client.
+        If no client is available a :py:class:`sentry_sdk.client.NoopClient` is returned.
+
+        .. versionadded:: X.X.X
+        """
+        current_scope = _current_scope.get()
+        if current_scope is not None and current_scope.client.is_active():
+            return current_scope.client
+
+        isolation_scope = _isolation_scope.get()
+        if isolation_scope is not None and isolation_scope.client.is_active():
+            return isolation_scope.client
+
+        if _global_scope is not None:
+            return _global_scope.client
+
+        return NoopClient()
+
+    def set_client(self, client=None):
+        # type: (Optional[sentry_sdk.client.BaseClient]) -> None
+        """
+        Sets the client for this scope.
+        :param client: The client to use in this scope.
+            If `None` the client of the scope will be replaced by a :py:class:`sentry_sdk.NoopClient`.
+
+        .. versionadded:: X.X.X
+        """
+        self.client = client or NoopClient()
+
+    @property
+    def is_forked(self):
+        # type: () -> bool
+        """
+        Whether this scope is a fork of another scope.
+
+        .. versionadded:: X.X.X
+        """
+        return self.original_scope is not None
+
+    def fork(self):
+        # type: () -> Scope
+        """
+        Returns a fork of this scope.
+
+        .. versionadded:: X.X.X
+        """
+        forked_scope = copy(self)
+        forked_scope.original_scope = self
+
+        return forked_scope
+
+    def isolate(self):
+        # type: () -> None
+        """
+        Creates a new isolation scope for this scope.
+        The new isolation scope will be a fork of the current isolation scope.
+
+        .. versionadded:: X.X.X
+        """
+        isolation_scope = Scope.get_isolation_scope()
+        forked_isolation_scope = isolation_scope.fork()
+        _isolation_scope.set(forked_isolation_scope)
```

Comment:
I'm fuzzy on the difference between this and the `isolated_scope` context manager. When would you use one or the other? I see the slight difference in `isolated_scope` also taking into account the current scope, but in the other PR we're always doing scope merge of all the scopes so not sure when this makes a difference.

Reply:
will try to come up with some use cases in existing integrations. If there are none I am happy to remove this